src/main/java/org/opentosca/container/client/impl/SwaggerContainerClient.java

```diff
@@ -78,8 +63,11 @@ public CompletableFuture<List<Application>> getApplicationsAsync() {
         executor.submit(() -> {
             List<Application> applications = new ArrayList<>();
             try {
-                for (CsarDTO csar : this.client.getCsars().getCsars()) {
-                    applications.add(getApplication(csar.getId()).orElseThrow(IllegalStateException::new));
+                CsarListDTO csarListDTO = this.client.getCsars();
+                if (csarListDTO.getCsars() != null) {
+                    for (CsarDTO csar : this.client.getCsars().getCsars()) {
```

Comment:
I would get it from `csarListDTO` since you already have the list.

Reply:
fixed